src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -231,7 +231,8 @@ public TransactionSubstate Run(EvmState state, IWorldState worldState, ITxTracer
                         (IReadOnlyCollection<Address>)currentState.DestroyList,
                         (IReadOnlyCollection<LogEntry>)currentState.Logs,
                         callResult.ShouldRevert,
-                        _txTracer != NullTxTracer.Instance);
+                        // Check ShouldRevert first as it is less expensive than IsTracing
+                        isTracerConnected: callResult.ShouldRevert && _txTracer.IsTracing);
```

Comment:
Is it a big deal?

Reply:
Is so many interfaces calls; can just implement it rather than having it as default implementation